src/dune_engine/scheduler.ml

```diff
@@ -1078,16 +1207,13 @@ module Run = struct
     Option.iter watcher ~f:(fun watcher ->
         ignore (wait_for_process t (Dune_file_watcher.pid watcher) : _ Fiber.t));
     ignore (kill_and_wait_for_all_processes t : saw_signal);
+    if Lazy.is_val t.timer then Timer.close (Lazy.force t.timer);
```

Comment:
Aren't we using `go` multiple times in some cases (in tests?). I'm worried we'll use the timer after close or (which is weirder) force the lazy value after close.

Reply:
> Aren't we using go multiple times in some cases (in tests?)

We're using a new scheduler each time, so I don't see how it's a problem.

>  I'm worried we'll use the timer after close or (which is weirder) force the lazy value after close.

Creating a timer after close is already a code error. We still allow cancelling timers after close however. Should that be allowed?

Reply:
Fair enough, no need to change anything then.